stdlib/LinearAlgebra/src/LinearAlgebra.jl

```diff
@@ -190,10 +190,45 @@ abstract type Algorithm end
 struct DivideAndConquer <: Algorithm end
 struct QRIteration <: Algorithm end
 
+# Pivoting strategies for matrix factorization algorithms.
 abstract type PivotingStrategy end
+
+"""
+    NoPivot
+
+Pivoting is not performed. Matrix factorizations such as the LU factorization
+may fail without pivoting.
+"""
 struct NoPivot <: PivotingStrategy end
+
+"""
+    RowNonZero
+
+First non-zero element in the remaining rows is chosen as the pivot element.
+
+Note that the [element type](@ref eltype) of the matrix must support the [`iszero`](@ref)
+method.
+"""
 struct RowNonZero <: PivotingStrategy end
+
+"""
+    RowMaximum
+
+The maximum element in the remaining rows is chosen as the pivot element.
+
+Note that the [element type](@ref eltype) of the matrix must support the [`abs`](@ref)
+and [`<`](@ref) methods.
```

Comment:
This refers to the return type of `abs`, not the eltype of the matrix. I'm not sure how to reword this, but as a proposal take

```suggestion
Note that the [element type](@ref eltype) of the matrix must admit an [`abs`](@ref) method,
whose result type must admit a [`<`](@ref) method.
```

Reply:
Hello @dkarrasch, I fully agree with your comments, esp. regarding `<`, which was an oversight. I'll incorporate all suggestions. Thank you very much for your support!